resources/js/features/game-list/components/GameListItems/GameListItemDrawerBacklogToggleButton.tsx

```diff
@@ -1,72 +1,43 @@
 import { useLaravelReactI18n } from 'laravel-react-i18n';
 import type { FC } from 'react';
-import { useEffect, useState } from 'react';
 import { MdClose } from 'react-icons/md';
 
 import { BaseButton } from '@/common/components/+vendor/BaseButton';
-import { usePageProps } from '@/common/hooks/usePageProps';
-import { useWantToPlayGamesList } from '@/common/hooks/useWantToPlayGamesList';
 import { cn } from '@/utils/cn';
 
 import { useDelayedButtonDisable } from './useDelayedButtonDisable';
+import { useGameBacklogState } from './useGameBacklogState';
 
 interface GameListItemDrawerBacklogToggleButtonProps {
   game: App.Platform.Data.Game;
   isInBacklog: boolean;
-
-  onToggle?: (newValue: boolean) => unknown;
 }
 
 export const GameListItemDrawerBacklogToggleButton: FC<
   GameListItemDrawerBacklogToggleButtonProps
-> = ({ game, isInBacklog, onToggle }) => {
-  const { auth } = usePageProps();
-
+> = ({ game, isInBacklog }) => {
   const { t } = useLaravelReactI18n();
 
-  const { addToWantToPlayGamesList, isPending, removeFromWantToPlayGamesList } =
-    useWantToPlayGamesList();
-
-  // We want to change the icon instantly for the user, even if the mutation is still running.
-  const [isInBacklogOptimistic, setIsInBacklogOptimistic] = useState(isInBacklog);
+  const {
+    isPending,
+    toggleBacklog,
+    isInBacklogMaybeOptimistic: isInBacklogOptimistic,
+  } = useGameBacklogState({
+    game: game,
+    isInitiallyInBacklog: isInBacklog,
+    shouldShowToasts: false,
+  });
 
   const isButtonDisabled = useDelayedButtonDisable(isPending);
 
-  // When the actual `isInBacklog` changes, update the optimistic state accordingly.
-  useEffect(() => {
-    setIsInBacklogOptimistic(isInBacklog);
-  }, [isInBacklog]);
-
-  const handleToggleFromBacklogClick = () => {
-    if (!auth?.user && typeof window !== 'undefined') {
-      window.location.href = route('login');
-
-      return;
-    }
-
-    const newValue = !isInBacklogOptimistic;
-
-    setIsInBacklogOptimistic(newValue);
-    onToggle?.(newValue);
-
-    const mutationPromise = isInBacklog
-      ? removeFromWantToPlayGamesList(game.id, game.title, { shouldEnableToast: false })
-      : addToWantToPlayGamesList(game.id, game.title, { shouldEnableToast: false });
-
-    mutationPromise.catch(() => {
-      setIsInBacklogOptimistic(isInBacklog);
-      onToggle?.(isInBacklog);
-    });
-  };
-
   return (
     <BaseButton
       variant="secondary"
       className={cn(
         'border disabled:!pointer-events-auto disabled:!opacity-100',
         isInBacklogOptimistic ? '!border-red-500 !bg-embed !text-red-500' : 'border-transparent',
       )}
-      onClick={handleToggleFromBacklogClick}
+      onClick={toggleBacklog}
```

Comment:
Clicking remove from the popup all the Want to Play list (in mobile) does not remove the item from the list, nor does it change the X to a plus.

Similarly, adding a game from the popup on the All Games list (in mobile) does not change the plus to an X. Furthermore, clicking on the plus generates a Something Went Wrong error.

Reply:
Thanks for catching that.

I've made some changes in latest to reduce duplicate backlog states, as well as improve this mobile UX.

Now, clicking remove from the drawer in the Want to Play list cleanly closes the drawer, then removes the list item and pops a toast.

I've double-checked table rows, list item rows, and the drawer toggle on all games & the want to play list.